subgraph/src/public-lock.ts

```diff
@@ -320,3 +337,63 @@ export function handleLockMetadata(event: LockMetadataEvent): void {
     lock.save()
   }
 }
+
+/**
+ * Create Receipt object for subgraph for key 'purchase'/'extend'/'renewal'
+ * @param {String} keyID - key id
+ * @param {event} Object - Object event
+ * @return {void}
+ */
+export function createReceipt(event: ethereum.Event): void {
+  const lockAddress = event.address.toHexString()
+  const hash = event.transaction.hash.toHexString()
+
+  const receipt = new Receipt(hash)
+
+  const lock = Lock.load(lockAddress)
+
+  const tokenAddress =
+    lock && lock.tokenAddress ? lock.tokenAddress : Bytes.fromHexString('')
+
+  if (lock && tokenAddress.toString().length > 0) {
+    const txReceipt = event.receipt!
+    const logs: ethereum.Log[] = txReceipt.logs
+    if (logs) {
+      for (let i = 0; i < logs.length; i++) {
+        const txLog = logs[i]
+        const ERC20_TRANSFER_TOPIC0 = Bytes.fromByteArray(
+          crypto.keccak256(
+            Bytes.fromHexString('Transfer(address,address.uint256')
```

Comment:
Are you confident this is correct here? `'Transfer(address,address.uint256'` is not an hex string.